be/src/util/json.cpp

```diff
@@ -216,6 +219,21 @@ int64_t JsonValue::hash() const {
     return to_vslice().normalizedHash();
 }
 
+// NOTE
+// Returned Json pointer should not be deconstructed!!
+// Memory allocated by binary_ is located at mempool, not in std::malloc, which should not
+// be freeed by deconstructor.
+// Instead, the memory should be freeed by the mempool
+StatusOr<JsonValue*> JsonValue::clone(MemPool* mem) const {
+    DCHECK_NOTNULL(mem);
+
+    pooled_string binary(this->binary_, pool_allocator<char>(mem));
+    JsonValue* res = (JsonValue*)mem->allocate(sizeof(JsonValue));
```

Comment:
add a static_assert(std::is_trivial_v<JsonValue>) to ensure JsonValue was a POD type.

Reply:
It seems JsonValue was not a POD type. so we need to call the default constructor here

Reply:
Next line is the in-place constructor for JsonValue